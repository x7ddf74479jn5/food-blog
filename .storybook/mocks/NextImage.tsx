/**
 * @see next/image mock
 * https://github.com/vercel/next.js/issues/18393#issuecomment-955577890
 */

import React from "react";
import * as NextImage from "next/future/image";

const OriginalNextImage = NextImage.default;

Object.defineProperty(NextImage, "default", {
  configurable: true,
  value: (/** @type {import('next/image').ImageProps} */ props) => {
    if (typeof props.src === "string") {
      return <OriginalNextImage {...props} unoptimized blurDataURL={props.src} />;
    } else {
      // don't need blurDataURL here since it is already defined on the StaticImport type
      return <OriginalNextImage {...props} unoptimized />;
    }
  },
});
