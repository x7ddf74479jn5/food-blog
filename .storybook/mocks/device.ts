export const device = {
  desktop: "PC HD",
  mobile: "iPhone12",
  tablet: "iPad Pro 12.9-in",
};
