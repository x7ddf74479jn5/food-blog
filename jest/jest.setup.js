// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
