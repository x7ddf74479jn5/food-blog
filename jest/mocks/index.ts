export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
