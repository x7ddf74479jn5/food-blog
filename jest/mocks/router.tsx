import type { AppRouterInstance, HooksClientContext } from "mocks/next/router";
import { AppRouterContext, createMockRouterHoc } from "mocks/next/router";

export const defaultMockRouter: AppRouterInstance = {
  back: jest.fn(),
  forward: jest.fn(),
  prefetch: jest.fn(),
  push: jest.fn(),
  refresh: jest.fn(),
  replace: jest.fn(),
};

export const withMockRouter = createMockRouterHoc({
  initialMockRouter: defaultMockRouter,
});

export { AppRouterContext };
export type { AppRouterInstance, HooksClientContext };
