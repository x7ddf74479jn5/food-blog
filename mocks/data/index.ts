export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
