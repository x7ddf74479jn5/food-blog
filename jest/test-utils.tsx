import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
