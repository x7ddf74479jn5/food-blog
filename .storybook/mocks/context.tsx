import { defaultMockRouter } from "mocks/next/router";
import { AppRouterContext } from "mocks/next/router";
import { SearchProvider } from "@/components/organisms/SearchArea/SearchContext";
import { createMockRouterHoc } from "mocks/next/router";
import React from "react";

export const withContext = (Story: React.ReactElement) => {
  return (
    <AppRouterContext.Provider value={defaultMockRouter}>
      <SearchProvider>{Story}</SearchProvider>
    </AppRouterContext.Provider>
  );
};

export const withRouterContext = createMockRouterHoc();
