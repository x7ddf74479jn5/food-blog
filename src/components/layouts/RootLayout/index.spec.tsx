import { withMockedRouter } from "jest/test-utils";
import { mockCategories, mockConfig } from "mocks/data";
import renderer from "react-test-renderer";

import { RootLayout } from ".";

describe("components/layouts/RootLayout", () => {
  const categoryList = Object.values(mockCategories);
  it("snapshot", () => {
    const tree = renderer
      .create(
        withMockedRouter(
          { asPath: "/" },
          <RootLayout categories={categoryList} config={mockConfig}>
            children
          </RootLayout>
        )
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
