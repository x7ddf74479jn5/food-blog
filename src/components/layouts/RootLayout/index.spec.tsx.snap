// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layouts/RootLayout snapshot 1`] = `
<div
  className="flex flex-col px-4 mx-auto mb-16 max-w-screen-lg min-h-screen"
>
  <header
    className="py-2"
  >
    <div
      className="flex justify-between items-center"
    >
      <a
        className="text-2xl font-bold text-green-500"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Food Blog
      </a>
      <div
        className="hidden sm:block w-1/3"
      >
        <div
          className="relative"
        >
          <div
            className="flex items-center py-2 px-3 space-x-2 bg-gray-100 dark:bg-gray-700 rounded-full"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "gray",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
            <input
              className="w-full dark:text-gray-100 bg-gray-100 dark:bg-gray-700 focus:outline-none"
              inputMode="text"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Search..."
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="flex items-center space-x-4"
      />
    </div>
    <div
      className="block sm:hidden mt-2"
    >
      <div
        className="relative"
      >
        <div
          className="flex items-center py-2 px-3 space-x-2 bg-gray-100 dark:bg-gray-700 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "gray",
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            className="w-full dark:text-gray-100 bg-gray-100 dark:bg-gray-700 focus:outline-none"
            inputMode="text"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyPress={[Function]}
            placeholder="Search..."
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
  </header>
  children
</div>
`;
